engine/tests/ElggCoreFilestoreTest.php

```diff
@@ -48,6 +48,25 @@ public function testFilenameOnFilestore() {
 		$this->assertFalse(file_exists($filepath));
 	}
 
+	public function testPathFromDataRoot() {
+
+		// create a user to own the file
+		$user = $this->createTestUser();
+		$dir = new \Elgg\EntityDirLocator($user->guid);
+
+		// setup a test file
+		$file = new \ElggFile();
+		$file->owner_guid = $user->guid;
+		$file->setFilename('testing/filestore.txt');
+		$file->open('write');
+		$file->write('Testing!');
+		$this->assertTrue($file->close());
+
+		$this->assertIdentical($file->getPathFromDataRoot(), $dir . 'testing/filestore.txt');
```

Comment:
This test is incomplete, we also would need to test `is_file(elgg_get_data_root() . $file->getPathFromDataRoot())`